test/jdk/java/awt/Robot/ScreenCaptureRobotTest.java

```diff
@@ -0,0 +1,162 @@
+ * @run main ScreenCaptureRobotTest
+ */
+public class ScreenCaptureRobotTest {
+
+    private static long delay = 500;
+    private static Frame frame;
+    private static volatile Canvas canvas;
+    private static BufferedImage realImage;
+    private static BufferedImage displayImage;
+    private static int difference;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            initializeGUI();
+            doTest();
+        } finally {
+            EventQueue.invokeAndWait(ScreenCaptureRobotTest::disposeFrame);
+        }
+    }
+
+    private static void initializeGUI() {
+        frame = new Frame("Test frame");
+        realImage = GraphicsEnvironment.getLocalGraphicsEnvironment()
+                .getDefaultScreenDevice().getDefaultConfiguration()
+                .createCompatibleImage(200, 100);
+
+        Graphics g = realImage.createGraphics();
+        g.setColor(Color.yellow);
+        g.fillRect(0, 0, 200, 100);
+        g.setColor(Color.red);
+        g.setFont(new Font("SansSerif", Font.BOLD, 30));
+        g.drawString("Capture This", 10, 40);
+        g.dispose();
+        displayImage = realImage;
+
+        canvas = new ImageCanvas();
+        canvas.setBackground(Color.YELLOW);
+        frame.setLayout(new BorderLayout());
+        frame.add(canvas);
+        frame.setSize(300, 200);
+        frame.setLocation(100, 100);
+        frame.setVisible(true);
+        canvas.requestFocus();
+    }
+
+    private static void doTest() throws Exception {
+        Robot robot;
+        robot = new Robot();
+        Thread.sleep(delay);
+
+        Point pnt = canvas.getLocationOnScreen();
+        Rectangle rect = new Rectangle(pnt.x + 10, pnt.y + 10, 200, 100);
+
+        // Capturing Image using Robot
+        BufferedImage capturedImage = robot.createScreenCapture(rect);
+
+        if (!compareImages(capturedImage, realImage)) {
+            String errorMessage = "FAIL : Captured Image is different from "
+                    + "the actual image by " + difference + " pixels";
+            System.err.println("Test failed");
+            throw new RuntimeException(errorMessage);
+        }
+        System.out.println("Test passed");
+    }
+
+    private static boolean compareImages(BufferedImage capturedImg,
+            BufferedImage realImg) throws InterruptedException {
+        int[] capturedPixels;
+        int[] realPixels;
+        int imgWidth;
+        int imgHeight;
+        int toleranceLevel = 0;
+        boolean result = true;
+
+        imgWidth = capturedImg.getWidth(null);
+        imgHeight = capturedImg.getHeight(null);
+
+        capturedPixels = new int[imgWidth * imgHeight];
+        realPixels = new int[imgWidth * imgHeight];
+
+        // Extracting Pixels from Captured Image
+        PixelGrabber pgCapturedImg = new PixelGrabber(capturedImg, 0, 0,
+                imgWidth, imgHeight, capturedPixels, 0, imgWidth);
+        pgCapturedImg.grabPixels();
```

Comment:
PixelGrabber ? You have a BufferedImage. Just get the pixels directly.

Reply:
Tried to get pixels directly from BufferedImage.
Got the RGB values of each pixel and compared them.
Hope this approach is ok.